extensions/query-store/src/settings/configureDialog.ts

```diff
@@ -0,0 +1,626 @@
+			.withProps({
+				name: constants.basedOnLabel,
+				label: constants.avgLabel
+			}).component();
+
+		this.criteriaBasisMaxRadioButton = this._view.modelBuilder.radioButton()
+			.withProps({
+				name: constants.basedOnLabel,
+				label: constants.maxLabel
+			}).component();
+
+		this.criteriaBasisMinRadioButton = this._view.modelBuilder.radioButton()
+			.withProps({
+				name: constants.basedOnLabel,
+				label: constants.minLabel
+			}).component();
+
+		this.criteriaBasisStdDevRadioButton = this._view.modelBuilder.radioButton()
+			.withProps({
+				name: constants.basedOnLabel,
+				label: constants.stdDevLabel
+			}).component();
+
+		this.criteriaBasisTotalRadioButton = this._view.modelBuilder.radioButton()
+			.withProps({
+				name: constants.basedOnLabel,
+				label: constants.totalLabel
+			}).component();
+
+		const basedOnLabel = this._view.modelBuilder.text()
+			.withProps({
+				value: constants.basedOnLabel
+			}).component();
+
+		this.criteriaBasisTotalRadioButton.checked = true;
+
+		let flexRadioButtonsModel = this._view.modelBuilder.flexContainer()
+			.withLayout({ flexFlow: 'column' })
+			.withItems([basedOnLabel, this.criteriaBasisAvgRadioButton, this.criteriaBasisMaxRadioButton, this.criteriaBasisMinRadioButton, this.criteriaBasisStdDevRadioButton, this.criteriaBasisTotalRadioButton])
+			.withProps({ ariaRole: 'radiogroup' })
+			.component();
+
+		return {
+			component: flexRadioButtonsModel
+		};
+	}
+
+	private createShowChartComponent(): azdata.FormComponent {
+		this.executionCountCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.executionCountLabel
+			}).component();
+
+		this.durationCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.durationLabel
+			}).component();
+
+		this.CPUTimeCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.CPUTimeLabel
+			}).component();
+
+		this.logicalReadsCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.logicalReadsLabel
+			}).component();
+
+		this.logicalWritesCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.logicalWritesLabel
+			}).component();
+
+		this.physicalReadsCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.physicalReadsLabel
+			}).component();
+
+		this.CLRTimeCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.CLRTimeLabel
+			}).component();
+
+		this.DOPCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.DOPLabel
+			}).component();
+
+		this.memoryConsumptionCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.memoryConsumptionLabel
+			}).component();
+
+		this.rowCountCheckBox = this._view.modelBuilder.checkBox()
+			.withProps({
+				label: constants.rowCountLabel
+			}).component();
+
+		this.executionCountCheckBox.checked = true;
```

Comment:
is the state of these saved somewhere? Does the dialog remember if one of these options is changed, applied, then the config dialog is reopened?

Reply:
can you also add a todo comment in the code?

Reply:
Will do. Also, added a work item to track it.